third_party/txt/src/txt/paragraph_builder.cc

```diff
@@ -77,11 +77,32 @@ void ParagraphBuilder::AddText(const char* text) {
   AddText(u16_text);
 }
 
+void ParagraphBuilder::AddPlaceholder(PlaceholderRun& span) {
+  obj_replacement_char_indexes_.insert(text_.size());
+  runs_.StartRun(PeekStyleIndex(), text_.size());
+  AddText(std::u16string(1ull, objReplacementChar));
+  runs_.StartRun(PeekStyleIndex(), text_.size());
+  inline_placeholders_.push_back(span);
+}
+
 std::unique_ptr<Paragraph> ParagraphBuilder::Build() {
   runs_.EndRunIfNeeded(text_.size());
 
+  // When placeholders are added, we replace all non-placeholder instances of
+  // 0xFFFC with 0xFFFD as we reserve 0xFFFC to indicate the position of
+  // placeholders.
+  if (!obj_replacement_char_indexes_.empty()) {
```

Comment:
Can the indexes be passed on to the Paragraph instead of rewriting all the objReplacementChars here?